### DISCUSSION TITLE
Unwanted curved edges while using react flow dagre.js

### DISCUSSION THREAD

**Discussion by gswebdev07** [xyflow/xyflow#3218] [Layouting]
I have been using dagre.js library to auto layout the nodes for my particular requirement. Expected behaviour is that whenever connecting two nodes which are on straight path a straight edge must be rendered but instead a stepdown like curved edge is seen between nodes. 
`dagreGraph.setGraph({
        rankdir: 'LR',
        nodesep: 40,
        edgesep: 0,
        ranksep: 100,
        ranker: 'longest-path',
        align: 'UL',
    })
`
    
It is to be noted that when is use align 'UR' the curve  is not seen and a straight path is rendered but my usecase requires me to  use 'UL'

In the example below you can see that the edge with index 2 is curved.

![Screenshot 2023-07-12 at 9 33 19 PM](https://github.com/wbkd/react-flow/assets/70875359/bd9de7a2-c0f6-49d4-b261-7a265898ac0b)

**Comment by chrtze** [comment]
Hey, if I understand your question correctly, you can just set type: "straight" on your edge object?

**Comment by DylanMerigaud** [comment]
The stepdown appears because the smoothstep elbow sits at the midpoint between the two handles, and each handle is at its node's center. When two nodes on a "straight" run have even slightly different heights their centers differ, so the connector kinks. That's why `align: 'UR'` happens to line them up and `'UL'` doesn't.

Two ways out: for a genuinely straight single-in / single-out chain, snap the child's cross-axis center onto the parent's so the run is colinear regardless of dagre's `align`; or just `type: "straight"` as suggested if you don't need the stepped look. I ended up putting the colinear-snap into a small dagre wrapper in case it's handy: https://www.npmjs.com/package/react-flow-auto-layout
